xz 5.6.4

ELundby45 opened the pull request:
xz 5.6.4

**Destination channel:** defaults

### Links

- [https://anaconda.atlassian.net/browse/PKG-6941](https://anaconda.atlassian.net/browse/PKG-6941) 
- [Upstream repository](https://github.com/tukaani-project/xz/tree/v5.6.4)
- [Upstream changelog/diff](https://github.com/tukaani-project/xz/releases/tag/v5.6.4)
### Explanation of changes:

- Port windows build to cmake. Since the VS Project files[ are no longer provided](https://github.com/tukaani-project/xz/blob/4fdcbfaf3f222299747c6a815762a74eeb1b0b23/windows/INSTALL-MSVC.txt#L24-L26). 
- ~`liblzma_static.lib` is now `lzma_static.lib` see [this comment](https://github.com/tukaani-project/xz/blob/4fdcbfaf3f222299747c6a815762a74eeb1b0b23/CMakeLists.txt#L1571-L1580)~
- Port unix to cmake.
- Stop building static libs.

ELundby45 committed:
Update meta.yaml

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_ddvi7z2fip/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:60: invalid_url: https://xz.tukaani.org/xz-utils/ : HTTPSConnectionPool(host='xz.tukaani.org', port=443): Max retries exceeded with url: /xz-utils/ (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0xffffb2622630>: Failed to resolve 'xz.tukaani.org' ([Errno -2] Name or service not known)"))
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

ELundby45 committed:
Update url

ELundby45 committed:
v5.6.4-win

https://github.com/tukaani-project/xz/blob/4fdcbfaf3f222299747c6a815762a74eeb1b0b23/CMakeLists.txt#L1571-L1580

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_35dqadequg/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:59: invalid_url: https://xz.tukaani.org/xz-utils/ : HTTPSConnectionPool(host='xz.tukaani.org', port=443): Max retries exceeded with url: /xz-utils/ (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0xffffaaa9f230>: Failed to resolve 'xz.tukaani.org' ([Errno -2] Name or service not known)"))
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

ELundby45 committed:
Update urls

JeanChristopheMorinPerso commented on recipe/bld.bat:
```diff
@@ -1,35 +1,33 @@
+rm -rf ./*
+
+:: Build static library
+cmake %CMAKE_COMMON_ARGS% -DBUILD_SHARED_LIBS=OFF ..
```
This could be a good occasion to drop the static library. We would need to do it on all architectures though.

ELundby45 committed:
Don't build static libs.

ELundby45 committed:
White space

danpetry commented on recipe/meta.yaml:
```diff
@@ -41,14 +38,11 @@ test:
     - test -f ${PREFIX}/include/lzma.h                          # [unix]
     - test -f ${PREFIX}/lib/pkgconfig/liblzma.pc                # [unix]
     - test -f `pkg-config --variable=libdir --dont-define-prefix liblzma`/liblzma${SHLIB_EXT}  # [unix]
-    - test -f ${PREFIX}/lib/liblzma.a                           # [unix]
```
could make sure these don't exist?

danpetry commented on recipe/meta.yaml:
```diff
@@ -41,14 +38,11 @@ test:
     - test -f ${PREFIX}/include/lzma.h                          # [unix]
     - test -f ${PREFIX}/lib/pkgconfig/liblzma.pc                # [unix]
     - test -f `pkg-config --variable=libdir --dont-define-prefix liblzma`/liblzma${SHLIB_EXT}  # [unix]
```
does it still produce .pc files if it's built with cmake? I guess so, or these tests would fail

danpetry replied:
yes, it does, but maybe there should be similar tests for cmake here too

danpetry reviewed (commented):
couple of comments, looks good in general

ELundby45 committed:
Review feedback.

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -39,16 +36,16 @@ test:
     - xz.exe --help    # [win]
     - unxz.exe --help  # [win]
     - test -f ${PREFIX}/include/lzma.h                          # [unix]
+    - test -f ${PREFIX}/lib/cmake/liblzma/liblzma-targets.cmake # [unix]
```
I would recommend to test `liblzma-config.cmake` instead of `targets` since `targets` is kind of an "implementation" detail. Basically, `targets` gets loaded by `config`. What cmake really searches for is the `config` file, not the `targets`.

ELundby45 committed:
Feedback.
